### ISSUE TITLE
[streams-platform] AddonProvider sets the streams:addon value in routes. This results in empty control panel sections.

### ISSUE THREAD

**Issue by RobinRadic** [pyrocms/pyrocms#5031]
**Is your feature request related to a problem? Please describe.**
When adding routes in a module `ex1-module` for another module `users-module`, for example when adding a section/form/table, its not possible to add these routes to the service providers `$routes` property. This is because `AddonProvider::registerRoutes` adds/overrides the streams:addon value to $addon->getNamespace(). This would result in empty control panel sections.

So the routes need to be defined in the map() method. This way the control panel sections for the users-module are shown

_See screenshots below_ + [code examples](https://github.com/pyradic/dev/tree/609cba101d2561a761880292b70f0a5820d75539/addons/shared/examples/ex1-module#routing)

**Describe the solution you'd like**
On [`AddonProvider@339`](https://github.com/anomalylabs/streams-platform/blob/1.7/src/Addon/AddonProvider.php#L339)
Would it be possible to implement something like:
```php
if(!array_has($route, 'streams::addon')){
    array_set($route, 'streams::addon', $addon->getNamespace());
}
```

This way, in `$routes` you can define `streams:addon` yourself like
```php
    protected $routes = [
        '/admin/users/departments'           => [
            'as'   => 'examples.module.ex1::departments.index',
            'uses' => 'Examples\Ex1Module\Http\Controller\Admin\DepartmentsController@index',
            'streams::addon' => 'anomaly.module.users'
        ],
   ];
```

**Comment by RobinRadic** [comment]
Some visuals as clarification:
When navigating to `/admin/users`, using either the `$routes` or `map()` way of defining the routes:
![users-module](https://user-images.githubusercontent.com/754732/65840765-19ec8b00-e31d-11e9-8ede-1603508f5b4a.png)

----------


When navigating to `/admin/users/departments`, using the `$routes` way of defining the routes:
![users-module-incorrect](https://user-images.githubusercontent.com/754732/65840722-b06c7c80-e31c-11e9-9110-dd190750abf6.png)

----------

When navigating to `/admin/users/departments`, using the `map()` way of defining the routes:
![user-module-correct](https://user-images.githubusercontent.com/754732/65840721-afd3e600-e31c-11e9-86aa-6237a565dae0.png)

**Comment by fryiee** [comment]
@RyanThompson @RobinRadic I'm going to take a run at this if it's agreeable?

**Comment by fryiee** [comment]
@RobinRadic this was implemented in 1.7/1.8 (tagged at 0d20ad3af8780731e6fca6cd205662bb30b37cdb and 050d017d6c3e252a84862eccf8b7cc5545679b31)
